light/odr.go

```diff
@@ -119,6 +119,23 @@ func (req *BlockRequest) StoreResult(db ethdb.Database) {
 	rawdb.WriteBodyRLP(db, req.Hash, req.Number, req.Rlp)
 }
 
+type hashOrNumber struct {
+	Hash   common.Hash
+	Number uint64
+}
+
+type HeaderRequest struct {
+	OdrRequest
+	Origin hashOrNumber
+	Header *types.Header
+}
+
+func (req *HeaderRequest) StoreResult(db ethdb.Database) {
```

Comment:
Should this function verify the headers before saving it to the chaindb (https://github.com/celo-org/celo-blockchain/blob/master/consensus/istanbul/backend/engine.go#L104)?

Reply:
Actually, looks like this header verification function should be used: https://github.com/celo-org/celo-blockchain/blob/master/consensus/istanbul/backend/engine.go#L221.

Reply:
Currently this is done in `client_handler` via `lightchain.InsertHeaderChain` which calls `headerchain.insertHeaderChain` which calls `engine.verifyHeaders`
`Request.Validate` doesn't have a hook to the engine object from what I see so I don't know if I can do this here